code/components/citizen-devtools/src/ResourceTimeWarnings.cpp

```diff
@@ -139,6 +201,27 @@ static InitFunction initFunction([]()
 			}
 			ImGui::End();
 			ImGui::PopFont();
+		}
+		else if ((usec() - lastHitch) < 5s)
+		{
+			ImGui::PushFont(GetConsoleFontTiny());
+
+			const float DISTANCE = 10.0f;
+			ImVec2 window_pos = ImVec2(ImGui::GetMainViewport()->Pos.x + ImGui::GetIO().DisplaySize.x - DISTANCE, ImGui::GetMainViewport()->Pos.y + ImGui::GetIO().DisplaySize.y - DISTANCE);
+			ImVec2 window_pos_pivot = ImVec2(1.0f, 1.0f);
+			ImGui::SetNextWindowPos(window_pos, ImGuiCond_Always, window_pos_pivot);
+
+			ImGui::SetNextWindowBgAlpha(0.3f); // Transparent background
+			if (ImGui::Begin("Time Warning", NULL, ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoResize | ImGuiWindowFlags_AlwaysAutoResize | ImGuiWindowFlags_NoSavedSettings | ImGuiWindowFlags_NoFocusOnAppearing | ImGuiWindowFlags_NoNav))
+			{
+				ImGui::Text(va("/!\\ %s", gettext("Slow system performance detected")));
+				ImGui::Separator();
+				ImGui::Text("%s", gettext("A call into the Windows API took too long recently and led to a game stutter.").c_str());
```

Comment:
Maybe too technical?

Reply:
Possibly, but also straight to the point and makes it pretty clear it isn't an issue with FiveM or the server, so I think it should be fine.

Reply:
Yeah - that's the point I was trying to make. 'Game stutter' hopefully is interpretable enough, people make up weird names for this.